include/flatbuffers/util.h

```diff
@@ -45,13 +50,63 @@
 
 namespace flatbuffers {
 
+#ifdef FLATBUFFERS_PREFER_PRINTF
+template<typename T> size_t IntToDigitCount(T t) {
+  size_t digit_count = 0;
+
+  // Count the sign for negative numbers
+  if (t < 0) {
+    digit_count++;
+  }
+
+  // Count a single 0 left of the dot for fractional numbers
+  if (-1 < t && t < 1) {
+    digit_count++;
+  }
+
+  // Count digits until fractional part
+  T eps = std::numeric_limits<float>::epsilon();
+  while (t <= (-1 + eps) || (1 - eps) <= t) {
+    t /= 10;
+    digit_count++;
+  }
+
+  return digit_count;
+}
+
+template<typename T> size_t NumToStringWidth(T t, int precision = 0) {
+  size_t string_width = IntToDigitCount(t);
+  if (precision) {
```

Comment:
this doesn't look correct.. if precision == 0, this means it wants default precision for the type, not 0. I think the if can be removed, and also remove the default arguments `= 0` from these functions.

Reply:
How does a user specify 0 precision then? Right now the `precision = 0` serves for the integer types as well.